smart/plugin.go

```diff
@@ -97,13 +116,58 @@ func validateName(namespace []string) bool {
 	return true
 }
 
+// Function to check properness of configuration parameters
+// and set plugin attribute accordingly
+func (sc *SmartCollector) setProcDevPath(cfg interface{}) error {
+	sc.initializedMutex.Lock()
+	defer sc.initializedMutex.Unlock()
+	procPath, err := config.GetConfigItem(cfg, "proc_path")
+	if err == nil && len(procPath.(string)) > 0 {
+		procPathStats, err := os.Stat(procPath.(string))
+		if err != nil {
+			return err
+		}
+		if !procPathStats.IsDir() {
+			return errors.New(fmt.Sprintf("%s is not a directory", procPath.(string)))
+		}
+		sc.proc_path = procPath.(string)
+	}
+	devPath, err := config.GetConfigItem(cfg, "dev_path")
+	if err == nil && len(devPath.(string)) > 0 {
+		devPathStats, err := os.Stat(devPath.(string))
+		if err != nil {
+			return err
+		}
+		if !devPathStats.IsDir() {
+			return errors.New(fmt.Sprintf("%s is not a directory", devPath.(string)))
+		}
+		sc.dev_path = devPath.(string)
+	}
+	if sysUtilProvider == nil {
+		sysUtilProvider = NewSysutilProvider(sc.proc_path, sc.dev_path)
+	}
+	sc.initialized = true
+	return nil
+}
+
 type SmartCollector struct {
+	initialized      bool
+	initializedMutex *sync.Mutex
+	logger           *log.Logger
+	proc_path        string
+	dev_path         string
 }
 
 type smartResults map[string]interface{}
 
 // CollectMetrics returns metrics from smart
 func (sc *SmartCollector) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, error) {
+	if !sc.initialized {
```

Comment:
@lmroz no pb at all, fully agree that it is better to do it right at the soonest.
Current modification on its way for new review